### PR TITLE
Add Grapheme segmenter that is a part of UAX29.

### PR DESCRIPTION
**Pull request by makotokato**

### CODE REVIEW

File: experimental/segmenter/data/grapheme.toml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+# GB11
+left = [ "Extended_Pictographic_Extend_ZWJ" ]
+right = [ "Extended_Pictographic" ]
+break_state = false
+
+[[rules]]
+# GB12/13
+left = [ "RI_RI" ]
+right = [ "Regional_Indicator" ]
+break_state = true
```
Comment (aethanyc) [inline experimental/segmenter/data/grapheme.toml:199]: Question: Trying to understand the GB12 & GB13 rules [1]. The spec says: "That is, do not break between regional indicator (RI) symbols if there is an odd number of RI characters before the break point."

Does it mean that if there are three RIs, it's ok to break between the second and third RI? Looks like this rule describes such a break opportunity.

[1] https://www.unicode.org/reports/tr29/#GB12

**Review by aethanyc — APPROVED** [review]
Good good. I have some inline nits.

File: experimental/segmenter/data/grapheme.toml (hunk 1 above)
Comment (makotokato) [reply experimental/segmenter/data/grapheme.toml:199]: GB12/13 is that RI RI RI should be RI RI ÷ RI. Twice RI marks as combine state, so twice RI doesn't break. So this rules defines it.

File: experimental/segmenter/data/grapheme.toml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+break_state = false
+
+[[rules]]
+# GB4/GB5
+left = [ "Control", "CR", "LF" ]
+right = [ "Control", "CR", "Extend", "L", "LV", "LVT", "V", "T" ]
+break_state = true
+
+[[rules]]
+left = [ "Control", "LF" ]
```
[the remark this answers is not in the document]
Comment (makotokato) [reply experimental/segmenter/data/grapheme.toml:108]: Thanks.

File: experimental/segmenter/src/grapheme.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
+use core::str::CharIndices;
+
+use crate::break_iterator_impl;
+use crate::indices::{Latin1Indices, Utf16Indices};
+use crate::rule_segmenter::*;
+
+include!(concat!(env!("OUT_DIR"), "/generated_grapheme_table.rs"));
+
+// UTF-8 version of word break iterator using rule based segmenter.
```
[the remark this answers is not in the document]
Comment (makotokato) [reply experimental/segmenter/src/grapheme.rs:9]: Thanks.

**Review by aethanyc — APPROVED** [review]

**Pull request merged**